recipes/apriltag/all/conanfile.py

```diff
@@ -1,46 +1,63 @@
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration
 import os
-import stat
-from conans import ConanFile, tools, CMake, AutoToolsBuildEnvironment
-from conans.errors import ConanException,ConanInvalidConfiguration
 
-class apriltagConan(ConanFile):
+
+class ApriltagConan(ConanFile):
     name = "apriltag"
-    license = "BSD-2-Clause"
-    homepage = "https://april.eecs.umich.edu/software/apriltag"
-    url = "https://github.com/conan-io/conan-center-index"
     description = ("AprilTag is a visual fiducial system, useful for a wide variety of tasks \
                     including augmented reality, robotics, and camera calibration")
+    homepage = "https://april.eecs.umich.edu/software/apriltag"
+    topics = ("conan", "apriltag", "robotics")
+    license = "BSD-2-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
     settings = "os", "arch", "compiler", "build_type"
-    options = {"shared": [True, False], "fPIC": [True, False]}
-    default_options = {'shared': False, "fPIC": True}
+
     generators = "cmake"
-    topics = ("conan", "apriltag", "robotics")
-    exports_sources = ["CMakeLists.txt", "patches/*"]
+    exports_sources = "CMakeLists.txt", "patches/*"
+
     _cmake = None
 
     @property
     def _source_subfolder(self):
         return "source_subfolder"
 
-    def _configure_cmake(self):
-        if not self._cmake:
-            self._cmake = CMake(self)
-            self._cmake.configure()
-        return self._cmake
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
 
     def configure(self):
-        if self.settings.os != "Linux":
-            raise ConanInvalidConfiguration("Apriltag officially supported only on Linux")
+        if self.options.shared:
+            del self.options.fPIC
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
 
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("Apriltag officially supported only on Linux")
+
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
 
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure()
```

Comment:
Out of source builds?

Reply:
I don't know of a conan general way to include `conanbuildinfo.cmake`.
```cmake
include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
conan_basic_setup()
```
will fail.

It would be nice if conan passes an argument (e.g. `CONAN_INSTALL_FOLDER`) to `cmake` so one could do:

```cmake
include("${CONAN_INSTALL_FOLDER}/conanbuildinfo.cmake")
conan_basic_setup()
```